cabal-install/tests/UnitTests/Distribution/Client/Init/NonInteractive.hs

```diff
@@ -1265,3 +1271,35 @@ testGo label f g h inputs = testCase label $
   case (_runPrompt $ f emptyFlags) (NEL.fromList inputs) of
     Left  x -> g x
     Right x -> h x
+
+cliParserTests :: TestTree
+cliParserTests = testGroup "cli parser"
+  [ testCase "Parse single dependency" $ do
+      flags <- assumeAllParse $ commandParseArgs initCommand False ["-d", "base"]
+      flags @?= dummyFlags
+        { dependencies = fmap mkStringyDep ["base"]
+        }
+  , testCase "Parse multiple dependency flags" $ do
+      flags <- assumeAllParse $ commandParseArgs initCommand False
+        ["-d", "base", "-d", "vector"]
+      flags @?= dummyFlags
+        { dependencies = fmap mkStringyDep ["base", "vector"]
+        }
+  , testCase "Parse list of dependencies" $ do
+      flags <- assumeAllParse $ commandParseArgs initCommand False
+        ["-d", "base,vector"]
```

Comment:
Sneakily introduced new feature.

Reply:
I'd rename the PR to "Fix and improve" and keep it.

Reply:
and document in many places...

Reply:
Any idea where this might be documented?

Reply:
No idea where to document that, the flag doesn't seem to be documented in the .rst.

Reply:
Well, anywhere. In a changelog for starters. Perhaps in the `--help` output? Would it be awkward a sentence into the .rst?

Reply:
in the .rst, the options aren't explained at all. Thus, adding docs for only a subset of flags feels like a bad decision to me (at least I resist documenting the whole command right now :) ). Changelog and help message make sense, thank you very much!

Reply:
Done!